### PR TITLE
Implement delete functionality for attendances

### LINKED ISSUE #40: As a user I can delete an attendance entry
so that I can have updated attendance records

### PR DESCRIPTION
**Pull request by casaarlai**
image
image
image

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -12,6 +12,7 @@ public class CliSyntax {
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
     public static final Prefix PREFIX_ATTENDANCE = new Prefix("att/");
+    public static final Prefix PREFIX_ATTENDANCE_DELETE = new Prefix("attd/");
```
Comment (brennalaurentan) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:15]: I like the prefix you chose for the new command

File: src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public AttendanceDeleteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args,
+                PREFIX_ATTENDANCE_DATE);
+
+        Set<Index> indexes;
+        try {
+            indexes = ParserUtil.parseIndexes(List.of(argMultimap.getPreamble().split(" ")));
+        } catch (IllegalValueException ive) {
```
Comment (brennalaurentan) [inline src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java:35]: I like how you created a catch for this exception!

File: src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ */
+public class AttendanceDeleteCommandParser implements Parser<AttendanceDeleteCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AttendanceDeleteCommand
+     * and returns a AttendanceDeleteCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AttendanceDeleteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
```
Comment (brennalaurentan) [inline src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java:28]: I like how you made it a requirement for the parameter to be a non-null value

File: src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
```
Comment (brennalaurentan) [inline src/main/java/seedu/address/logic/parser/AttendanceDeleteCommandParser.java:3]: I like how you ordered the imports in lexicographical order

File: src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Person createEditedPerson(Person personToEdit, Attendance attendance) {
+        assert personToEdit != null;
+
+        Name updatedName = personToEdit.getName();
```
Comment (brennalaurentan) [inline src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java:92]: I like how neatly you structured your code here!

**Review by brennalaurentan — APPROVED** [review]
New delete attendance records feature LGTM! Nicely done

File: src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java (hunk 5 above)
Comment (casaarlai) [reply src/main/java/seedu/address/logic/commands/AttendanceDeleteCommand.java:92]: I am so flattered by all your positive comments! This spurs me on to work harder on our tP!

**Pull request merged**